Upgrade SDK to 6.0.100-rc.1.21411.13

ViktorHofer opened the pull request:
Both the AzDO and the Core-Eng team believe believe that the issue is on our side and was caused by a thread pool regression. The assumption stands that we need to update to a newer SDK which contains the fix for the thread pool hang.

Pros:
-	AzDO and Core-Eng believe that this will mitigate the AzDO feed restore issues.
Cons:
-	We will upgrade to an unsigned SDK build. Arcade and other repos already did the same to workaround the issue.
-	That SDK build isn’t officially released and won’t until RC 1 ships. This means that developers need to install that build via the nightly channel (from https://github.com/dotnet/installer) if they want to use their globally installed SDK in combination with dotnet/runtime.
-	Even though this is a breaking change, we can’t wait for the next monthly infrastructure rollout.

lewing committed:
set StaticWebAssetsEnabled=false

ViktorHofer committed:
Upgrade SDK to 6.0.100-rc.1.21379.2

Both the AzDO and the Core-Eng team believe believe that the issue is on our side and was caused by a thread pool regression. The assumption stands that we need to update to a newer SDK which contains the fix for the thread pool hang.

Pros:
-	AzDO and Core-Eng believe that this will mitigate the AzDO feed restore issues.
Cons:
-	We will upgrade to an unsigned SDK build. Arcade and other repos already did the same to workaround the issue.
-	That SDK build isn’t officially released and won’t until RC 1 ships. This means that developers need to install that build via the nightly channel (from https://github.com/dotnet/installer) if they want to use their globally installed SDK in combination with dotnet/runtime.
-	Even though this is a breaking change, we can’t wait for the next monthly infrastructure rollout.

stephentoub commented:
> Both the AzDO and the Core-Eng team believe believe that the issue is on our side and was caused by a thread pool regression.

Meaning https://github.com/dotnet/runtime/pull/56346?

If that's true, code somewhere is doing sync-over-async blocking waiting for a task.  Do they know where?  Can we fix that as well?

ViktorHofer commented:
> Meaning #56346?

Correct.

> If that's true, code somewhere is doing sync-over-async blocking waiting for a task. Do they know where? Can we fix that as well?

@MattGal where you the one who was in contact with AzDO about the issue?

jkoritzinsky commented:
My guess would be the sync-over-async is somewhere near the border of MSBuild and NuGet since MSBuild tasks can't be async.

MattGal commented:
> @MattGal where you the one who was in contact with AzDO about the issue?

Yes, and "normal" types of failure (HTTP 429, 503, etc) are likely still worth pursuing with them, since they'll have telemetry.

After much investigation though, it seemed like the connections weren't even getting through, and after a lot of chatting with @wfurt , a change from @kouvel seemed relevant.  Empirically ASP has had significant improvement from taking the new SDK as well.

ViktorHofer commented:
cc @johnterickson (AzDO packaging)

ViktorHofer commented:
> .dotnet/sdk/6.0.100-rc.1.21379.2/Sdks/Microsoft.NET.Sdk.Razor/targets/Microsoft.NET.Sdk.Razor.StaticWebAssets.targets(425,5): error : (NETCORE_ENGINEERING_TELEMETRY=Build) System.IO.DirectoryNotFoundException: Could not find a part of the path '/__w/1/s/artifacts/obj/mono/BrowserDebugHost/Release/net6.0/StaticWebAssets.build.json'.
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String path, OpenFlags flags, Int32 mode) in System.Private.CoreLib.dll:token 0x60000d5+0x0
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize) in System.Private.CoreLib.dll:token 0x60000d9+0x31
   at System.IO.File.WriteAllBytes(String path, Byte[] bytes) in System.Private.CoreLib.dll:token 0x6005ccc+0x2b
   at Microsoft.AspNetCore.Razor.Tasks.GenerateStaticWebAssetsManifest.PersistManifest(StaticWebAssetsManifest manifest) in Microsoft.NET.Sdk.Razor.Tasks.dll:token 0x600018a+0x3b
   at Microsoft.AspNetCore.Razor.Tasks.GenerateStaticWebAssetsManifest.Execute() in Microsoft.NET.Sdk.Razor.Tasks.dll:token 0x6000187+0xeb

@steveisok @akoeplinger who would be the best person to look into this issue in the wasm legs?

steveisok commented:
> @steveisok @akoeplinger who would be the best person to look into this issue in the wasm legs?

/cc @lewing @radical

ViktorHofer commented:
> If that's true, code somewhere is doing sync-over-async blocking waiting for a task. Do they know where? Can we fix that as well?

@stephentoub we would need to approach the product team that we believe own the faulty code. If we believe that's NuGet we should figure out how to approach them. One way I could imagine would be to help them reviewing their code to find such patterns.

cc @SteveMCarroll @danmoseley

MattGal commented:
> > If that's true, code somewhere is doing sync-over-async blocking waiting for a task. Do they know where? Can we fix that as well?
> 
> @stephentoub we would need to approach the product team that we believe own the faulty code. If we believe that's NuGet, we should figure out how to approach them. One way I could imagine would be to help them reviewing their code to find such patterns.
> 
> cc @SteveMCarroll @danmoseley

I think you may be looking to engage with the last few git-blame-folks from https://github.com/NuGet/NuGet.Client/blob/0ae80f5495c8d6fad2ec6e2c708727675ff08fb6/src/NuGet.Core/NuGet.Protocol/HttpSource/HttpSource.cs#L321-L350 , though it's very infrequently updated.

lewing commented:
> > .dotnet/sdk/6.0.100-rc.1.21379.2/Sdks/Microsoft.NET.Sdk.Razor/targets/Microsoft.NET.Sdk.Razor.StaticWebAssets.targets(425,5): error : (NETCORE_ENGINEERING_TELEMETRY=Build) System.IO.DirectoryNotFoundException: Could not find a part of the path '/__w/1/s/artifacts/obj/mono/BrowserDebugHost/Release/net6.0/StaticWebAssets.build.json'.
> > at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String path, OpenFlags flags, Int32 mode) in System.Private.CoreLib.dll:token 0x60000d5+0x0
> > at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize) in System.Private.CoreLib.dll:token 0x60000d9+0x31
> > at System.IO.File.WriteAllBytes(String path, Byte[] bytes) in System.Private.CoreLib.dll:token 0x6005ccc+0x2b
> > at Microsoft.AspNetCore.Razor.Tasks.GenerateStaticWebAssetsManifest.PersistManifest(StaticWebAssetsManifest manifest) in Microsoft.NET.Sdk.Razor.Tasks.dll:token 0x600018a+0x3b
> > at Microsoft.AspNetCore.Razor.Tasks.GenerateStaticWebAssetsManifest.Execute() in Microsoft.NET.Sdk.Razor.Tasks.dll:token 0x6000187+0xeb
> 

@ViktorHofer the issue is https://github.com/dotnet/runtime/pull/56974#issuecomment-895656894 I'll pull the changes I added there here

stephentoub commented:
Do we have a dump from any of the processes that hung?

lewing commented:
@lambdageek why are browser AOT/EAT lanes failing on hotreload here?

lambdageek commented:
@lewing Did something change about how wasm tests are built& executed


Two possibilities:
1. Does this line still work:

   https://github.com/dotnet/runtime/blob/95147163dac477da5177f5c5402ae9b93feb5c89/src/libraries/System.Runtime.Loader/tests/System.Runtime.Loader.Tests.csproj#L10

2. Do the AOT/EAT tests enable the hot reload runtime component?

stephentoub commented on src/libraries/Common/tests/System/Net/Prerequisites/NetCoreServer/NetCoreServer.csproj:
```diff
@@ -5,6 +5,7 @@
     <AspNetCoreHostingModel>InProcess</AspNetCoreHostingModel>
     <OutputType>Exe</OutputType>
     <EnableDefaultCompileItems>false</EnableDefaultCompileItems>
+    <StaticWebAssetsEnabled>false</StaticWebAssetsEnabled>
```
What is this?

radical replied:
https://github.com/dotnet/runtime/pull/57143#issuecomment-896214913

lewing replied:
cc @javiercn

danmoseley replied:
If it's fixed in the SDK do we need to remove these now?

lewing replied:
yes, I'll revert the various workarounds

radical commented:
> @lewing Did something change about how wasm tests are built& executed in rc1?
> 
> Two possibilities:
> 
> 1. Does this line still work:
>    https://github.com/dotnet/runtime/blob/95147163dac477da5177f5c5402ae9b93feb5c89/src/libraries/System.Runtime.Loader/tests/System.Runtime.Loader.Tests.csproj#L10

It does seem to be passed to the app:
```
[20:11:48] info: console.info: Arguments: --setenv=DOTNET_MODIFIABLE_ASSEMBLIES=debug,--run,WasmTestRunner.dll,System.Runtime.Loader.Tests.dll,-notrait,category=IgnoreForCI,-notrait,category=OuterLoop,-notrait,category=failing
```

[This](https://github.com/dotnet/runtime/commit/2bdf376f9e060770e2c7bfdd2280ed0ac18688a2) was a recent `setenv` related fix, but that should be fine, AFAICS.

> 2. Do the AOT/EAT tests enable the hot reload runtime component?

[This](https://github.com/dotnet/runtime/blob/main/src/mono/wasm/build/WasmApp.Native.targets#L257-L259)?
```xml
  <ItemGroup Condition="'$(Configuration)' == 'Debug' and '@(_MonoComponent->Count())' == 0">
    <_MonoComponent Include="hot_reload;debugger" />
  </ItemGroup>
```

radical commented:
> Ah, but the EAT/AOT tests look like they're Release configuration.

that's the same for the regular tests too.

> I set the `EnableDebugInformation` property for the hot reload test assemblies so that they're still modifiable (but the main test assembly is built normally):
> 
> https://github.com/dotnet/runtime/blob/53b7ba9f1070182ef8381116ef58a32e2fa5901d/src/libraries/System.Runtime.Loader/tests/ApplyUpdate/Directory.Build.props#L16-L18
> 
> maybe we should set `_MonoComponent` explicitly for this test assembly?

Are runtime components used with these tests at all? IIUC, they are available only through workload packs. And looking at the binlog, I don't see any `RuntimeConfigManifest.targets` in there.

Either way, what changed in this PR that is causing the failure?

Also, (not necessarily in this PR) do we need to add some public item to allow setting/overriding these components?

ViktorHofer commented:
As the rest of the stack already moved onto the RC1 SDK we want to get this in asap. Otherwise we will be stuck with outdated versions of our dependencies, i.e. hotreload-utils (https://github.com/dotnet/runtime/pull/56974#issuecomment-896272406).

lambdageek commented:
I think we should just delete the IsSupported2 test .  It's calling `MetadataUpdater.IsSupported` which the IL linker rewrites to `false` when trimming is enabled.  The test is effectively tautological: if MetadataUpdater.IsSupported returns true we expect it to return true.

lewing committed:
Remove IsSupported2 since asserting it is true when it might not be

lewing commented:
I've removed IsSupported2 lets see how things look now.

lewing commented:
I'm not sure what is causing the coreclr failure but they seem to be choking on some questionable command line + argument parsing.

cc @danmoseley

lewing commented:
```
        > set CORE_ROOT=C:\h\w\B068099D\p
        > C:\h\w\B068099D\w\AE9E09B7\e\tracing\eventpipe\processinfo\processinfo\processinfo.cmd
    tracing\eventpipe\processinfo2\processinfo2\processinfo2.cmd [FAIL]
      Unhandled exception. System.Exception: CommandLine must match current process. Expected: C:\h\w\B068099D\p\corerun.exe C:\h\w\B068099D\w\AE9E09B7\e\tracing\eventpipe\processinfo2\processinfo2\processinfo2.dll, Received: C:\h\w\B068099D\p\corerun.exe C:\h\w\B068099D\w\AE9E09B7\e\tracing\eventpipe\processinfo2\processinfo2\-p C:\h\w\B068099D\w\AE9E09B7\e\tracing\eventpipe\processinfo2\processinfo2\System.Reflection.Metadata.MetadataUpdater.IsSupported=false C:\h\w\B068099D\w\AE9E09B7\e\tracing\eventpipe\processinfo2\processinfo2\processinfo2.dll (original: "C:\h\w\B068099D\p\corerun.exe"  -p "System.Reflection.Metadata.MetadataUpdater.IsSupported=false" processinfo2.dll )
         at Tracing.Tests.Common.Utils.Assert(Boolean predicate, String message)
         at Tracing.Tests.ProcessInfoValidation.ProcessInfoValidation.Main(String[] args)
```
It appears the normalizing code is appending the path to each arg and the test doesn't expect the -p ?

lewing commented:
cc @josalem

josalem commented:
Which leg is this failure is coming from? Why is the `-p ...` there in the first place? The test is only expecting `corerun test.dll` to be in the commandline. If this is a niche corner case, you can just disable the test for that leg. If someone added this to _all_ legs I'll need to update the test to handle additional components in the commandline.

TL;DR - this isn't a product issue, this is an issue with how the test is being invoked.

danmoseley commented:
also wasm
```
[20:49:06] fail: [FAIL] System.Text.Json.Serialization.Tests.CacheTests.JsonSerializerOptionsUpdateHandler_ClearingDoesntPreventSerialization
[20:49:06] info: System.TypeLoadException : Could not load type 'System.Text.Json.JsonSerializerOptionsUpdateHandler' from assembly 'System.Text.Json, Version=6.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'.
[20:49:06] info:    at System.Reflection.RuntimeAssembly.GetType(String , Boolean , Boolean ) in /_/src/mono/System.Private.CoreLib/src/System/Reflection/RuntimeAssembly.cs:line 268
[20:49:06] info:    at System.Text.Json.Serialization.Tests.CacheTests.JsonSerializerOptionsUpdateHandler_ClearingDoesntPreventSerialization() in /_/src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/CacheTests.cs:line 230
[20:49:06] info: --- End of stack trace from previous location ---
[21:03:08] info: Finished:    System.Text.Json.Tests.dll
```
Something got trimmed away -- @eerhardt?

agocke commented:
@josalem Looks like there are a few failing EventPipe tests in this PR. Could you take a look?

josalem commented:
@agocke they're all the same test. It's the `processinfo*` tests that are failing cause the tests started passing arguments other than the test DLL to `corerun`. I'm drafting a patch to handle that right now. I didn't think `corerun` could take arguments other than a DLL.

danmoseley commented:
Microsoft.Extensions.FileProviders.Physical.Tests failure will be fixed by https://github.com/dotnet/runtime/pull/57235

lewing committed:
Try enabling the runtimeconfig generator everywhere

lewing commented:
@fanyang-mono Can you take a look at the runtimeconfig related failures

eerhardt commented:
> [20:46:37] fail: [FAIL] System.Text.Tests.UTF7EncodingEncode.Encode_Basic(source: "\xffff", index: 0, count: 1, expected: [43, 47, 47, 56, 45])
[20:46:37] info: System.NotSupportedException : Support for UTF-7 is disabled. See https://aka.ms/dotnet-warnings/SYSLIB0001 for more information.

This is because of https://github.com/dotnet/sdk/pull/18812. When we trim an application, we automatically set `EnableUnsafeUTF7Encoding=false`. If the test needs UTF-7, it should be turned on explicitly in the test.

> [20:49:06] fail: [FAIL] System.Text.Json.Serialization.Tests.CacheTests.JsonSerializerOptionsUpdateHandler_ClearingDoesntPreventSerialization
[20:49:06] info: System.TypeLoadException : Could not load type 'System.Text.Json.JsonSerializerOptionsUpdateHandler' from assembly 'System.Text.Json, Version=6.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'.

This is because of https://github.com/dotnet/sdk/pull/18670 and https://github.com/dotnet/sdk/pull/18810. Same as above, we are setting `MetadataUpdaterSupport=false` in Release, trimmed apps. If the test needs that functionality, it should be set explicitly.

lewing commented:
@eerhardt why was it passing before?

lewing commented:
@eerhard i assume it needs unsafebinaryformatter too then? https://github.com/dotnet/runtime/blob/main/src/libraries/System.Text.Encoding/tests/System.Text.Encoding.Tests.csproj#L9

eerhardt commented:
> If the test needs UTF-7, it should be turned on explicitly in the test.

More info: It looks like we _already_ set it in the System.Text.Encoding.Tests:

https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/src/libraries/System.Text.Encoding/tests/System.Text.Encoding.Tests.csproj#L9

However, I put a bug into the SDK 😞. I have an extra space in the Condition here:

https://github.com/dotnet/sdk/blob/9d86f2620cdea8c74a7fd5e5b8316477df5270b4/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.ILLink.targets#L41

```xml
    <EnableUnsafeUTF7Encoding Condition="'$(EnableUnsafeUTF7Encoding )' == ''">false</EnableUnsafeUTF7Encoding >
```

Will send a PR to the SDK to fix this shortly. I can push a change here that will work-around the bug for now.

buyaa-n commented:
> wasm tests -- this has been deprecated for a while, did we change warnings to errors? @buyaa-n

I think the obsoletion warnings have always been `errors` in the runtime, but the test project has disabled `SYSLIB0001` explicitly, not sure why it is failing for mono

https://github.com/dotnet/runtime/blob/08123228999d8790c12cf83b4f922ba4de789668/src/libraries/System.Text.Encoding/tests/System.Text.Encoding.Tests.csproj#L8

lewing commented:
> > If the test needs UTF-7, it should be turned on explicitly in the test.
> 
> More info: It looks like we _already_ set it in the System.Text.Encoding.Tests:
> 
> https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/src/libraries/System.Text.Encoding/tests/System.Text.Encoding.Tests.csproj#L9
> 
> However, I put a bug into the SDK 😞. I have an extra space in the Condition here:
> 
> https://github.com/dotnet/sdk/blob/9d86f2620cdea8c74a7fd5e5b8316477df5270b4/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.ILLink.targets#L41
> 
> ```
>     <EnableUnsafeUTF7Encoding Condition="'$(EnableUnsafeUTF7Encoding )' == ''">false</EnableUnsafeUTF7Encoding >
> ```
> 
> Will send a PR to the SDK to fix this shortly. I can push a change here that will work-around the bug for now.

I saw it right after I said that PR is https://github.com/dotnet/sdk/pull/19677

lewing committed:
Revert part of runtimeconfig parser change

eerhardt committed:
Fix System.Text.Json tests to run with trimming by setting MetadataUpdaterSupport=true.

Work around SDK issue by overwriting EnableUnsafeUTF7Encoding in System.Text.Encoding

josalem commented:
Wrote up a quick patch that handles the runtime config switches being passed at the comamndline. Should I just push it to this PR to unblock things?

danmoseley commented:
@josalem yes please.

agocke committed:
Disable bundling tests on Mac due to bug

See https://github.com/dotnet/runtime/issues/57242

danmoseley commented:
We believe all known issues will be fixed once @josalem pushes his workaround. 

Then per @ViktorHofer recommendation we should rerun CI on this a few times to root out flakiness, then merge and send mail/post sticky issue to notify of the new min SDK baseline.

Commit:
Make ProcessInfo tests resilient to RuntimeConfig switches on commandline

lewing commented:
@josalem looks there is still an issue would also normalizing the original string as the comparison work?

lewing commented:
runtime (Mono monointerpreter Pri0 Runtime Tests Run OSX x64 release)  failed with https://github.com/dotnet/runtime/issues/11063 🤦

danmoseley commented:
We still have the UTF7Encoding issue.

A new failure in WASM:
```
XHarness command issued: wasm test-browser --app=. --output-directory=/datadisks/disk1/work/B57C0983/w/B6F609AF/uploads/xharness-output --engine=V8 --engine-arg=--stack-trace-limit=1000 --js-file=runtime.js -- --run WasmTestRunner.dll System.Diagnostics.Tools.Tests.dll -notrait category=IgnoreForCI -notrait category=OuterLoop -notrait category=failing
[01:16:23] info: Starting chromedriver with args: --headless --incognito --enable-features=NetworkService,NetworkServiceInProcess --allow-insecure-localhost --disable-background-timer-throttling --disable-backgrounding-occluded-windows --disable-breakpad --disable-component-extensions-with-background-pages --disable-dev-shm-usage --disable-extensions --disable-features=TranslateUI --disable-ipc-flooding-protection --disable-renderer-backgrounding --force-color-profile=srgb --metrics-recording-only
Connection refused [::ffff:127.0.0.1]:44539 (localhost:44539)
Connection refused [::ffff:127.0.0.1]:44539 (localhost:44539)
Connection refused [::ffff:127.0.0.1]:44539 (localhost:44539)
Connection refused [::ffff:127.0.0.1]:44539 (localhost:44539)
Connection refused [::ffff:127.0.0.1]:44539 (localhost:44539)
Connection refused [::ffff:127.0.0.1]:44539 (localhost:44539)
```

@agocke we still have a failure in Put_app_directory_behind_symlink_and_use_dotnet_run
https://dev.azure.com/dnceng/public/_build/results?buildId=1289153&view=ms.vss-test-web.build-test-results-tab&runId=38061862&paneView=debug&resultId=100348

and as a bonus item we have a segfault in Arm64 System.Text.Json.Tests ...

Commit:
Update global.json

Updating to 6.0.100-rc.1.21411.13

radical commented:
> A new failure in WASM:
> 

> Connection refused [::ffff:127.0.0.1]:44539 (localhost:44539)
> Connection refused [::ffff:127.0.0.1]:44539 (localhost:44539)
> Connection refused [::ffff:127.0.0.1]:44539 (localhost:44539)
> Connection refused [::ffff:127.0.0.1]:44539 (localhost:44539)
> Connection refused [::ffff:127.0.0.1]:44539 (localhost:44539)
> Connection refused [::ffff:127.0.0.1]:44539 (localhost:44539)


Chromedriver can take some time to launch chrome, and connect to it, and till it does, we get these messages. But in this particular case, it seems to be trying to for quite some time, and when chrome does actually start up, the driver side seems to have cancelled the operation. Maybe this was caused by the machine being too busy?

I'm retrying the build to see if it passes on a second run.

radical commented:
```
dotnet-install: Downloading primary link https://dotnetcli.azureedge.net/dotnet/Sdk/6.0.100-rc.1.21411.13/dotnet-sdk-6.0.100-rc.1.21411.13-linux-x64.tar.gz
curl: (22) The requested URL returned error: 404 
dotnet-install: The resource at https://dotnetcli.azureedge.net/dotnet/Sdk/6.0.100-rc.1.21411.13/dotnet-sdk-6.0.100-rc.1.21411.13-linux-x64.tar.gz is not available.
dotnet-install: Downloading legacy link https://dotnetcli.azureedge.net/dotnet/Sdk/6.0.100-rc.1.21411.13/dotnet-dev-ubuntu.18.04-x64.6.0.100-rc.1.21411.13.tar.gz
curl: (22) The requested URL returned error: 404 
dotnet-install: The resource at https://dotnetcli.azureedge.net/dotnet/Sdk/6.0.100-rc.1.21411.13/dotnet-dev-ubuntu.18.04-x64.6.0.100-rc.1.21411.13.tar.gz is not available.
dotnet_install: Error: Could not find `.NET Core SDK` with version = 6.0.100-rc.1.21411.13
```

eerhardt committed:
Update to latest SDK.
Remove System.Text.Encoding.Tests workaround

Commit:
ignore runtimeconfig switches _if_ they show up

josalem commented:
About to push an update to the test that should handle this (I had been testing on Linux, but it appears this only fails on Windows). That did point out that the logic in `SaveManagedCommandLine` or perhaps `GetCommandLine` in the PAL is behaving differently between Windows and non-Windows:
https://github.com/dotnet/runtime/blob/main/src/coreclr/vm/ceeload.cpp#L11365-L11411

On non-Windows, this is saving only `corerun app.dll`, but on Windows, it appears to be saving `corerun <all the property arguments> app.dll`.  I'm don't think this is a problem directly, but the difference in behavior is odd.

I'll push the update in a sec.

danmoseley commented:
I didn't pull down the dump for the JSON crash, which seems sporadic. If I have time later today I will try.

danmoseley commented:
@dotnet/dnceng what could cause there to be only a single line in the console output? (`Console log: 'JIT.jit64.hfa' from job 76fdf20e-5995-47a2-a5d4-656da646e9e1 workitem 9bdf57b5-ba46-4640-b90d-73652d96f040 (windows.10.arm64v8.open) executed on machine DDARM64-146`)

Seems a JIT test failed but there is no info.

https://dev.azure.com/dnceng/public/_build/results?buildId=1290308&view=ms.vss-test-web.build-test-results-tab&runId=38085910&resultId=103042&paneView=dotnet-dnceng.dnceng-build-release-tasks.helix-test-information-tab
https://helixre8s23ayyeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-pull-57143-merge-76fdf20e599547a2a5/JIT.jit64.hfa/1/console.2cf9b544.log?sv=2019-07-07&se=2021-09-01T16%3A19%3A55Z&sr=c&sp=rl&sig=EySUK0YGxErdTG%2Fj2fRKRzKmAE%2BhwwIQJxMoFQNgma0%3D

MattGal commented:
> @dotnet/dnceng what could cause there to be only a single line in the console output? (`Console log: 'JIT.jit64.hfa' from job 76fdf20e-5995-47a2-a5d4-656da646e9e1 workitem 9bdf57b5-ba46-4640-b90d-73652d96f040 (windows.10.arm64v8.open) executed on machine DDARM64-146`)
> 
> Seems a JIT test failed but there is no info.
> 
> https://dev.azure.com/dnceng/public/_build/results?buildId=1290308&view=ms.vss-test-web.build-test-results-tab&runId=38085910&resultId=103042&paneView=dotnet-dnceng.dnceng-build-release-tasks.helix-test-information-tab
> https://helixre8s23ayyeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-pull-57143-merge-76fdf20e599547a2a5/JIT.jit64.hfa/1/console.2cf9b544.log?sv=2019-07-07&se=2021-09-01T16%3A19%3A55Z&sr=c&sp=rl&sig=EySUK0YGxErdTG%2Fj2fRKRzKmAE%2BhwwIQJxMoFQNgma0%3D

Taking a peek

MattGal commented:
@danmoseley I am trying to get on the machine to directly check, but I _think_ what happened is the taskkill.exe call that is line 1 on the execute.cmd crashed it with -1073741502 (STATUS_DLL_INIT_FAILED) faster than it could record std output, which would indicate a funky machine.

I'll continue to investigate, but I'd also add you should now be able to safely remove the taskkill invocations entirely, as we now have reliable leaked process cleanup in the Helix client.

danmoseley commented:
@agocke here's more context for the failing test showing the similar passing tests.
```
    <collection total="13" passed="10" failed="1" skipped="2" name="Test collection for Microsoft.NET.HostModel.Tests.AppHostUsedWithSymbolicLinks" time="26.125">
      <test name="Microsoft.NET.HostModel.Tests.AppHostUsedWithSymbolicLinks.Run_framework_dependent_app_behind_symlink" type="Microsoft.NET.HostModel.Tests.AppHostUsedWithSymbolicLinks" method="Run_framework_dependent_app_behind_symlink" time="0" result="Skip">
        <reason><![CDATA[Currently failing in OSX with \"No such file or directory\" when running Command.Create. CI failing to use stat on symbolic links on Linux (permission denied).]]></reason>
      </test>
      <test name="Microsoft.NET.HostModel.Tests.AppHostUsedWithSymbolicLinks.Put_app_directory_behind_symlink_and_use_dotnet_run" type="Microsoft.NET.HostModel.Tests.AppHostUsedWithSymbolicLinks" method="Put_app_directory_behind_symlink_and_use_dotnet_run" time="5.8878282" result="Fail">
        <failure exception-type="Xunit.Sdk.XunitException">
          <message><![CDATA[Expected command to pass but it did not.\"\\nFile Name: /Users/runner/work/1/s/.dotnet/dotnet\\nArguments: run\\nExit Code: 150\\nStdOut:\\n\\nStdErr:\\nIt was not possible to find any compatible framework version\\nThe framework 'Microsoft.NETCore.App', version '6.0.0-rc.1.21411.2' was not found.\\n  - The following frameworks were found:\\n      2.1.0 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]\\n      2.1.1 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]\\n      2.1.2 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]\\n      2.1.3 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]\\n      2.1.4 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]\\n      2.1.5 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]\\n      2.1.6 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]\\n      2.1.7 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]\\n      2.1.8 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]\\n      2.1.9 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]\\n      2.1.10 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]\\n      2.1.11 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]\\n      2.1.12 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]\\n      2.1.13 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]\\n      2.1.14 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]\\n      2.1.15 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]\\n      2.1.16 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]\\n      2.1.17 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]\\n      2.1.18 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]\\n      2.1.19 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]\\n      2.1.20 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]\\n      2.1.21 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]\\n      2.1.22 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]\\n      2.1.23 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]\\n      2.1.24 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]\\n      2.1.25 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]\\n      2.1.26 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]\\n      2.1.27 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]\\n      2.1.28 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]\\n      3.1.0 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]\\n      3.1.1 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]\\n      3.1.2 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]\\n      3.1.3 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]\\n      3.1.4 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]\\n      3.1.5 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]\\n      3.1.6 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]\\n      3.1.7 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]\\n      3.1.8 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]\\n      3.1.9 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]\\n      3.1.10 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]\\n      3.1.11 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]\\n      3.1.12 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]\\n      3.1.13 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]\\n      3.1.14 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]\\n      3.1.15 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]\\n      3.1.16 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]\\n      3.1.17 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]\\n      5.0.0 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]\\n      5.0.1 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]\\n      5.0.2 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]\\n      5.0.3 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]\\n      5.0.4 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]\\n      5.0.5 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]\\n      5.0.6 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]\\n      5.0.7 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]\\n      5.0.8 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]\\n\\nYou can resolve the problem by installing the specified framework and/or SDK.\\n\\nThe specified framework can be found at:\\n  - https://aka.ms/dotnet-core-applaunch?framework=Microsoft.NETCore.App&framework_version=6.0.0-rc.1.21411.2&arch=x64&rid=osx.10.15-x64\\n\\n\"]]></message>
          <stack-trace><![CDATA[   at FluentAssertions.Execution.XUnit2TestFramework.Throw(String message)
   at FluentAssertions.Execution.TestFrameworkProvider.Throw(String message)
   at FluentAssertions.Execution.DefaultAssertionStrategy.HandleFailure(String message)
   at FluentAssertions.Execution.AssertionScope.FailWith(String message, Object[] args)
   at Microsoft.DotNet.CoreSetup.Test.CommandResultAssertions.Pass() in /_/src/installer/tests/TestUtils/Assertions/CommandResultAssertions.cs:line 30
   at Microsoft.NET.HostModel.Tests.AppHostUsedWithSymbolicLinks.Put_app_directory_behind_symlink_and_use_dotnet_run() in /_/src/installer/tests/Microsoft.NET.HostModel.Tests/Microsoft.NET.HostModel.AppHost.Tests/AppHostUsedWithSymbolicLinks.cs:line 207]]></stack-trace>
        </failure>
      </test>
      <test name="Microsoft.NET.HostModel.Tests.AppHostUsedWithSymbolicLinks.Run_apphost_behind_symlink(symlinkRelativePath: \&quot;a/b/SymlinkToApphost\&quot;)" type="Microsoft.NET.HostModel.Tests.AppHostUsedWithSymbolicLinks" method="Run_apphost_behind_symlink" time="2.1038309" result="Pass" />
      <test name="Microsoft.NET.HostModel.Tests.AppHostUsedWithSymbolicLinks.Run_apphost_behind_symlink(symlinkRelativePath: \&quot;a/SymlinkToApphost\&quot;)" type="Microsoft.NET.HostModel.Tests.AppHostUsedWithSymbolicLinks" method="Run_apphost_behind_symlink" time="1.7897959" result="Pass" />
      <test name="Microsoft.NET.HostModel.Tests.AppHostUsedWithSymbolicLinks.Put_dotnet_behind_symlink" type="Microsoft.NET.HostModel.Tests.AppHostUsedWithSymbolicLinks" method="Put_dotnet_behind_symlink" time="2.0057718" result="Pass" />
      <test name="Microsoft.NET.HostModel.Tests.AppHostUsedWithSymbolicLinks.Run_framework_dependent_app_with_runtime_behind_symlink" type="Microsoft.NET.HostModel.Tests.AppHostUsedWithSymbolicLinks" method="Run_framework_dependent_app_with_runtime_behind_symlink" time="0" result="Skip">
        <reason><![CDATA[Currently failing in OSX with \"No such file or directory\" when running Command.Create. CI failing to use stat on symbolic links on Linux (permission denied).]]></reason>
      </test>
      <test name="Microsoft.NET.HostModel.Tests.AppHostUsedWithSymbolicLinks.Put_satellite_assembly_behind_symlink" type="Microsoft.NET.HostModel.Tests.AppHostUsedWithSymbolicLinks" method="Put_satellite_assembly_behind_symlink" time="2.5457662" result="Pass" />
      <test name="Microsoft.NET.HostModel.Tests.AppHostUsedWithSymbolicLinks.Put_app_directory_behind_symlink" type="Microsoft.NET.HostModel.Tests.AppHostUsedWithSymbolicLinks" method="Put_app_directory_behind_symlink" time="2.2338867" result="Pass" />
      <test name="Microsoft.NET.HostModel.Tests.AppHostUsedWithSymbolicLinks.Put_app_directory_behind_symlink_and_use_dotnet" type="Microsoft.NET.HostModel.Tests.AppHostUsedWithSymbolicLinks" method="Put_app_directory_behind_symlink_and_use_dotnet" time="2.0881602" result="Pass" />
      <test name="Microsoft.NET.HostModel.Tests.AppHostUsedWithSymbolicLinks.Run_apphost_behind_transitive_symlinks(firstSymlinkRelativePath: \&quot;a/b/FirstSymlink\&quot;, secondSymlinkRelativePath: \&quot;c/d/SecondSymlink\&quot;)" type="Microsoft.NET.HostModel.Tests.AppHostUsedWithSymbolicLinks" method="Run_apphost_behind_transitive_symlinks" time="2.1247919" result="Pass" />
      <test name="Microsoft.NET.HostModel.Tests.AppHostUsedWithSymbolicLinks.Run_apphost_behind_transitive_symlinks(firstSymlinkRelativePath: \&quot;a/b/FirstSymlink\&quot;, secondSymlinkRelativePath: \&quot;c/SecondSymlink\&quot;)" type="Microsoft.NET.HostModel.Tests.AppHostUsedWithSymbolicLinks" method="Run_apphost_behind_transitive_symlinks" time="1.9710878" result="Pass" />
      <test name="Microsoft.NET.HostModel.Tests.AppHostUsedWithSymbolicLinks.Run_apphost_behind_transitive_symlinks(firstSymlinkRelativePath: \&quot;a/FirstSymlink\&quot;, secondSymlinkRelativePath: \&quot;c/d/SecondSymlink\&quot;)" type="Microsoft.NET.HostModel.Tests.AppHostUsedWithSymbolicLinks" method="Run_apphost_behind_transitive_symlinks" time="1.7784765" result="Pass" />
      <test name="Microsoft.NET.HostModel.Tests.AppHostUsedWithSymbolicLinks.Run_apphost_behind_transitive_symlinks(firstSymlinkRelativePath: \&quot;a/FirstSymlink\&quot;, secondSymlinkRelativePath: \&quot;c/SecondSymlink\&quot;)" type="Microsoft.NET.HostModel.Tests.AppHostUsedWithSymbolicLinks" method="Run_apphost_behind_transitive_symlinks" time="1.5951218" result="Pass" />
    </collection>
```

agocke commented:
I'll take a look -- I didn't see that failure the first time

ViktorHofer commented:
Without wanting to put too much pressure on us, Arcade's test retry feature is currently blocked on getting this in. So ideally we get this in today so that we can update our dependencies later today or tomorrow.

agocke committed:
Remove test which uses `dotnet run, which is SDK-specific

agocke commented:
I've removed the test -- it's dependent on SDK-specific functionality (`dotnet run`) so I don't think it should be in the runtime at all. I think the cause of that error is infra, specifically version mismatches in the livebuild layout vs installer tests. I think it's a low value test that we shouldn't block on.

danmoseley commented:
@dotnet/jit-contrib could one of you comment on this? https://github.com/dotnet/runtime/pull/57143#issuecomment-897840269
I cannot find where "execute.cmd" exists or is created.

More importantly, are you comfortable with us merging with that failure, or should we resolve it first? We are pushing on this SDK ingestion because it should improve infra reliability.

MattGal commented:
> @dotnet/jit-contrib could one of you comment on this? [#57143 (comment)](https://github.com/dotnet/runtime/pull/57143#issuecomment-897840269)
> I cannot find where "execute.cmd" exists or is created.
> 
> More importantly, are you comfortable with us merging with that failure, or should we resolve it first? We are pushing on this SDK ingestion because it should improve infra reliability.

Execute.cmd is auto-generated by the Arcade SDK.  Check out HelixPre/PostCommands in your projects.

e.g.
src\coreclr\scripts\superpmi.proj
src\libraries\sendtohelixhelp.proj
src\tests\Common\helixpublishwitharcade.proj

ViktorHofer commented:
Right, https://github.com/dotnet/runtime/search?q=taskkill lists the ones that Matt just listed.

> More importantly, are you comfortable with us merging with that failure, or should we resolve it first? We are pushing on this SDK ingestion because it should improve infra reliability.

Sounds like that was a faulty machine state and not related to this change.

MattGal commented:
> Right, https://github.com/dotnet/runtime/search?q=taskkill lists the ones that Matt just posted.
> 
> > More importantly, are you comfortable with us merging with that failure, or should we resolve it first? We are pushing on this SDK ingestion because it should improve infra reliability.
> 
> Sounds like that was a faulty machine state and likely isn't related to this change.

While it does still seem like a weird crash happened, whatever it was didn't reproduce with the same payload on the same machine (I really wanted to make sure since I'm pushing this update).  It does use the updated SDK for test execution, but I saw no evidence this is caused by the update.

ViktorHofer commented:
I see a couple Installer issues just popped up. @agocke can you please take a look?

agocke commented:
@ViktorHofer These weren't in the previous runs. Did something change?

AndyAyersMS commented:
> @dotnet/jit-contrib could one of you comment on this? [#57143 (comment)](https://github.com/dotnet/runtime/pull/57143#issuecomment-897840269)
> I cannot find where "execute.cmd" exists or is created.
> 
> More importantly, are you comfortable with us merging with that failure, or should we resolve it first? We are pushing on this SDK ingestion because it should improve infra reliability.

Looks like the tests passed and the crash happened later? 

I'd be fine merging with this.

danmoseley commented:
Thanks folks for the team effort merging this, let's see whether it helps CI stability.

ViktorHofer commented:
Amazing 👋👋👋
